scipy/stats/_boost/setup.py

```diff
@@ -0,0 +1,56 @@
+import pathlib
+import sys
+
+
+def pre_build_hook(build_ext, ext):
+    from scipy._build_utils.compiler_helper import get_cxx_std_flag
+    std_flag = get_cxx_std_flag(build_ext._cxx_compiler)
+    if std_flag is not None:
+        ext.extra_compile_args.append(std_flag)
+
+
+def configuration(parent_package='', top_path=None):
+    from scipy._lib._boost_utils import _boost_dir
+    from _info import _klass_mapper  # type: ignore
+    from numpy.distutils.misc_util import Configuration
+    import numpy as np
+    config = Configuration('_boost', parent_package, top_path)
+
+    DEFINES = [
+        # return nan instead of throwing
+        ('BOOST_MATH_DOMAIN_ERROR_POLICY', 'ignore_error'),
+    ]
+    if sys.maxsize > 2**32:
+        # 32-bit machines lose too much precision with no promotion,
+        # so only set this policy for 64-bit machines
```

Comment:
I looked this up, macro explanation is: _"Determines whether double types get promoted to long double internally to ensure maximum precision in the result, defaults to true, but can be set to false to turn promotion of double's off. "_

So with the way this is set up now, the idea is that `double` on 64-bit platforms does not get promoted to `longdouble` in order to improve performance - but on 32-bit platforms the regular `double` loops have worse and therefore promotion to `longdouble` must be done?

Reply:
> So with the way this is set up now, the idea is that `double` on 64-bit platforms does not get promoted to `longdouble` in order to improve performance - but on 32-bit platforms the regular `double` loops have worse and therefore promotion to `longdouble` must be done?

This is correct and the reason why we did this.  If this turns out to be an issue we can return to the default promotion policy and adjusting tolerances in tests as needed